### DISCUSSION TITLE
Can we replace setup.py by a Pipfile?

### DISCUSSION THREAD

**Discussion by HannoSpreeuw** [transientskp/tkp#599] [General]
I mostly develop Python applications and run these in a virtual environment, which I set up using pipenv.

tkp uses a setup.py. Can I replace that by a Pipfile?
[This SO discussion](https://stackoverflow.com/questions/47227406/pipenv-vs-setup-py) states that 
_If you're building an application, then pipenv is the only thing you need._

So is TraP an application or a library?

I would say it is an application and we could replace setup.py by a Pipfile.
But that may cause problems that I cannot foresee. 

First we may need to establish if installing TraP presently causes issues for many users and if a Pipfile would take away most of those issues, because [dependency versions are pinned down by pipenv](https://pipenv.pypa.io/en/latest/advanced/#pipfile-vs-setup-py).

**Comment by AntoniaR** [comment]
TraP is both an application and a library. We run the TraP pipeline to populate the TraP database, but then use the TraP database tools to access the database from separate scripts. Therefore, I don't think pipfile will do what we need.

Currently (Python 2 version), users find TraP very easy to install and run.
